drivers/vulkan/vulkan_context.cpp

```diff
@@ -210,55 +210,65 @@ VkBool32 VulkanContext::_check_layers(uint32_t check_count, const char **check_n
-	const char *instance_validation_layers_alt3[] = { "VK_LAYER_GOOGLE_threading", "VK_LAYER_LUNARG_parameter_validation", "VK_LAYER_LUNARG_object_tracker", "VK_LAYER_LUNARG_core_validation", "VK_LAYER_GOOGLE_unique_objects" };
+	uint32_t instance_layer_count;
 
-	uint32_t instance_layer_count = 0;
 	err = vkEnumerateInstanceLayerProperties(&instance_layer_count, nullptr);
-	ERR_FAIL_COND_V(err, ERR_CANT_CREATE);
+	if (err) {
+		ERR_FAIL_V(ERR_CANT_CREATE);
+	}
+
+	if (instance_layer_count < 1) {
+		return OK;
+	}
 
 	VkBool32 validation_found = 0;
 	uint32_t validation_layer_count = 0;
-	const char **instance_validation_layers = nullptr;
-	if (instance_layer_count > 0) {
-		VkLayerProperties *instance_layers = (VkLayerProperties *)malloc(sizeof(VkLayerProperties) * instance_layer_count);
-		err = vkEnumerateInstanceLayerProperties(&instance_layer_count, instance_layers);
-		if (err) {
-			free(instance_layers);
-			ERR_FAIL_V(ERR_CANT_CREATE);
-		}
+	const char *const *instance_validation_layers = nullptr;
 
-		validation_layer_count = ARRAY_SIZE(instance_validation_layers_alt1);
-		instance_validation_layers = instance_validation_layers_alt1;
-		validation_found = _check_layers(validation_layer_count, instance_validation_layers, instance_layer_count, instance_layers);
+	VkLayerProperties *instance_layers = (VkLayerProperties *)malloc(sizeof(VkLayerProperties) * instance_layer_count);
+	err = vkEnumerateInstanceLayerProperties(&instance_layer_count, instance_layers);
+	if (err) {
+		free(instance_layers);
+		ERR_FAIL_V(ERR_CANT_CREATE);
+	}
 
-		// use alternative (deprecated, removed in SDK 1.1.126.0) set of validation layers
-		if (!validation_found) {
-			validation_layer_count = ARRAY_SIZE(instance_validation_layers_alt2);
-			instance_validation_layers = instance_validation_layers_alt2;
-			validation_found = _check_layers(validation_layer_count, instance_validation_layers, instance_layer_count, instance_layers);
-		}
+	validation_layer_count = ARRAY_SIZE(instance_validation_layers_alt1);
+	instance_validation_layers = instance_validation_layers_alt1;
+	validation_found = _check_layers(validation_layer_count, instance_validation_layers, instance_layer_count, instance_layers);
 
-		// use alternative (deprecated, removed in SDK 1.1.121.1) set of validation layers
-		if (!validation_found) {
-			validation_layer_count = ARRAY_SIZE(instance_validation_layers_alt3);
-			instance_validation_layers = instance_validation_layers_alt3;
-			validation_found = _check_layers(validation_layer_count, instance_validation_layers, instance_layer_count, instance_layers);
-		}
+	// use alternative (deprecated, removed in SDK 1.1.126.0) set of validation layers
+	if (!validation_found) {
+		validation_layer_count = ARRAY_SIZE(instance_validation_layers_alt2);
+		instance_validation_layers = instance_validation_layers_alt2;
+		validation_found = _check_layers(validation_layer_count, instance_validation_layers, instance_layer_count, instance_layers);
+	}
```

Comment:
Seems this logic (and the one above and below) can be simplified (e.g: via loops).

Reply:
Done. Initially, I didn't want to refactor the code, but you're right about the simplification. LMK if you like the change.

Reply:
btw, github diff shows the changes in a very vague way.

Reply:
Looks good, thanks for the update!